### ISSUE TITLE
Multi Instance is broke after update

### ISSUE THREAD

**Issue by bboyaztec** [Inve1951/BetterDiscordStuff#197]
Discord pushed out a new update. Plugin is no longer functional.

**Comment by Inve1951** [comment]
Yes, but in fact this is not due to Discord's update but due to the electron-17 branch having been merged into the BetterDiscord 1.7.0 release, which removes direct NodeJS API access from the renderer process and polyfills most of what plugins use. Multi-Instance depends on the `child_process` API, which has not been polyfilled for security reasons.

I think that the functionality of Multi-Instance is something a plugin should be able to provide but as of now there is no fix and there probably won't be in the near future. I'll try to discuss this with Zerebos and Strencher again at some point - maybe we can come up with something.

Surely not as convenient but you can create a shortcut to `Discord.exe` (assuming Windows) and pass `--multi-instance` as an extra parameter. Running Discord from this shortcut will allow for a second window.

**Comment by bboyaztec** [comment]
Thanks for the shortcut tip. Good solution for my needs here.

**Comment by SmolAlli** [comment]
Also if you're interested, [OpenAsar](https://openasar.dev/) (which is compatible with BD) has multi-instance

**Comment by KiraFR** [comment]
I created a powershell script file which contains : 
`Start-Process -FilePath "DIR/TO/DISCORD/app-*/Discord.exe" -ArgumentList "--multi-instance"`
and i can execute it from my desktop

I prefered the script because after each update, the name of the folder "app-X.X.XXXX" change, so with this script, it detect automatically the exe

Tell me if there is a better solution :D

**Comment by auralicia** [comment]
> Also if you're interested, [OpenAsar](https://openasar.dev/) (which is compatible with BD) has multi-instance

hello, cna u explain more on how to use it? and which should i install? theres stable, PTB, and canary.

**Comment by clavzno** [comment]
> I created a powershell script file which contains : `Start-Process -FilePath "DIR/TO/DISCORD/app-*/Discord.exe" -ArgumentList "--multi-instance"` and i can execute it from my desktop
> 
> I prefered the script because after each update, the name of the folder "app-X.X.XXXX" change, so with this script, it detect automatically the exe
> 
> Tell me if there is a better solution :D

how does this work?

**Comment by TirOFlanc** [comment]
> Tell me if there is a better solution :D

`Start-Process -FilePath (Get-ChildItem "$env:localappdata\Discord\app-*\Discord.exe" | Select-Object -First 1).FullName -ArgumentList "--multi-instance"`


> how does this work?

You must paste this line into a txt file that you rename Biscord.ps1 (it's PowerShell)


But this technique no longer works.

**Comment by freejamp1302** [comment]
> > Tell me if there is a better solution :D
> 
> `Start-Process -FilePath (Get-ChildItem "$env:localappdata\Discord\app-*\Discord.exe" | Select-Object -First 1).FullName -ArgumentList "--multi-instance"`
> 
> > how does this work?
> 
> You must paste this line into a txt file that you rename Biscord.ps1 (it's PowerShell)
> 
> But this technique no longer works.

Is there another way to run multiple windows now?

**Comment by TirOFlanc** [comment]
> Is there another way to run multiple windows now?

You can actually run up to 3 or 4 accounts simultaneously with a simple combo:

Discord Stable (regular desktop app)
Discord PTB (Public Test Build) — download it from [discord.com/download](https://discord.com/download), scroll to the bottom of the page

Each desktop version is fully independent, so log in a different account on each.

Browser — open [discord.com](https://discord.com/) and log in a third account there
Browser profile — most browsers (Chrome, Edge, Firefox) let you create multiple profiles, each with its own session. Create a second profile, open discord.com in it, and you've got a fourth account running

**Comment by bscabl** [comment]
discord multi instance powershell script still works for me.
